galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/TestClassesRas.java

```diff
@@ -0,0 +1,93 @@
+
+package dev.galasa.framework.api.ras.internal;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.ras.RasTestClass;
+
+@Component(service = Servlet.class, scope = ServiceScope.PROTOTYPE, property = {
+        "osgi.http.whiteboard.servlet.pattern=/ras/testclasses" }, name = "TestClasses RAS")
+
+public class TestClassesRas extends HttpServlet {
+
+    /**
+     *
+     */
+    private static final long serialVersionUID = 1L;
+
+    private Gson gson = new Gson();
+    private final List<String> DUMMYQUERY = Arrays.asList("assc","ascc","aasc","ascbundle","descc","ddesc","deesc", "desc" );
+    private final List<RasTestClass> TESTCLASSDUMMY = Arrays.asList(new RasTestClass("FirstTest", "Abundle"),new RasTestClass("SecondTest", "BigBundle"),new RasTestClass("ThirdTest", "MiniBundle"),new RasTestClass("ZeroTest", "NanoBundle"));
+    // sorting options to search for in querry not testclasses, use equals intead of
+    // ==,
+    // throw servlet exception, not hide it
+    // error not needed, should return empty array with 200
+
+    @Reference
+    public IFramework framework; // NOSONAR
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        
+        JsonObject testclasses = new JsonObject();
+        JsonArray testObjects = new JsonArray();
+
+        for(String search: DUMMYQUERY){
```

Comment:
I am not sure what this is about.
The URL will be something like:-
/ras/testclasses?sort=testclass:asc,bundle:asc
So you need to parse that.   However Charlie has written a test class to do the parsing so you can use call that static class, to determine which sort technique is required.